### PR TITLE
[caffe2] pre-convert filter format in optimizeForIDEEP

### PR DESCRIPTION
**Pull request by gujinghui**
Convert the filters to expected format before computation.

### CODE REVIEW

**Comment by gujinghui** [comment]
@yinghai  pls help review
@jgong5

File: caffe2/opt/converter.cc [hunk 1]
```diff
@@ -11,7 +11,7 @@ using namespace nom;
 namespace {
 
 std::vector<int> getStrides(std::map<std::string, caffe2::Argument> argMap) {
-  std::vector<int> strides;
+  std::vector<int> strides = {1, 1};
```
Comment (yinghai) [inline caffe2/opt/converter.cc]: Why changing the default value?

Comment (gujinghui) [reply caffe2/opt/converter.cc]: to get the expected filter format, the expected_weights_descriptor API in optimize_ideep requires the init values for all parameters.

**Comment by ezyang** [comment]
ping @yinghai

**Comment by gujinghui** [comment]
@ezyang 
when #12164 and #12170, will update this one soon.

**Comment by gujinghui** [comment]
@yinghai  @ezyang 
Just rebased this PR.
Can we merge this one at first?
Thanks.

**Comment by yinghai** [comment]
Please split this into 3 PRs as you outlined above.

**Comment by gujinghui** [comment]
@yinghai 
split to #12970 and #12971.
pls merge those two first. Then, merge this PR.

**Comment by yinghai** [comment]
@gujinghui Is this still relevant?

**Review by yinghai — CHANGES_REQUESTED** [review]
No update for long.

**Comment by gujinghui** [comment]
@yinghai  Thanks for reminder.
Not need now. Will submit new PR for this if necessary.

**Pull request closed without merging**